### ISSUE TITLE
kube-prometheus change the k8s node internal-ip to public ip

### ISSUE THREAD

**Issue by wilean** [prometheus-operator/kube-prometheus#654]
Environment：
kube-prometheus v.0.6.0
kubernetes v1.18.6

before install:

```
kubectl get node -o wide 
NAME             STATUS   ROLES    AGE   VERSION   INTERNAL-IP    EXTERNAL-IP  
ali-k8s-master   Ready    master   16d   v1.18.6       10.81.50.195   <none>        
ali-k8s-node     Ready    <none>   5d    v1.18.6       10.46.227.29   <none>
```

but when i finish the installation:

```
kubectl get node -o wide 
NAME             STATUS   ROLES    AGE   VERSION   INTERNAL-IP    EXTERNAL-IP  
ali-k8s-master   Ready    master   16d   v1.18.6     10.81.50.195   <none>        
ali-k8s-node     Ready    <none>   5d    v1.18.6       47.97.45.65   <none>
```

It is a problem that the internal network ip of the node becomes the public network ip, which will cause many functions to malfunction.How to solve it？

**Comment by s-urbaniak** [comment]
Do you mind to elaborate on the actual question? I don't see anything wrong except the `-node` address being changed. Is that the problem you are referring to?

**Comment by wilean** [comment]
> Do you mind to elaborate on the actual question? I don't see anything wrong except the address being changed. Is that the problem you are referring to?`-node`

Yes, it is a problem that the internal network ip becomes the public network ip, which will cause many functions to malfunction

**Comment by s-urbaniak** [comment]
So far i don't see where in prometheus-operator manipulate node endpoints. How did you install prometheus-operator?

**Comment by wilean** [comment]
> So far i don't see where in prometheus-operator manipulate node endpoints. How did you install prometheus-operator?

I installed it like this：
```
cd kube-prometheus-0.6.0/
kubectl create -f manifests/setup
kubectl create -f manifests/
```
I did a test and only installed altermanager. If all three pods are on the same node, it will happen, and then reduce the number of pods to 1 to restore. But when I start other components, the internal network ip becomes public network ip again.
So I guess it may be that some components cannot run multiple on the same node, or there will be conflicts between components.
